### PR TITLE
sample: flow in pipeline

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

Sample for using flow in pipeline job as anonymous component.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/README.md
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/README.md:1]: gen github workflow for these new examples?

File: cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+type: pipeline
+
+name: simplePipelineJobWithFlow
+description: The hello world pipeline job with flow option 1
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/pipeline.yml:4]: what's option 1?

File: cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+type: pipeline
+
+name: simplePipelineJobWithFlow
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/pipeline.yml:3]: add dispaly name not name in example, make the name-with-this-format

File: cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+This is a dummy pipeline job with anonymous reference for flow as component. 
+
+Note that `flow.tools.json` must be generated before using a flow as a component. After that, You need to edit the compute cluster in the defaults section and run `az ml job create --file pipeline.yml` to submit the pipeline job.
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/README.md:3]: add link to flow repo and doc?

File: cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/web_classification/flow.dag.yaml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            logit_bias: ''
+            text: ${fetch_text_content_from_url.output}
+          provider: AzureOpenAI
+          connection: azure_open_ai_connection
+          api: completion
+          module: promptflow.tools.aoai
+environment:
+    python_requirements_txt: requirements.txt 
+additional_includes:
+  - ../additional_includes/convert_to_dict.py
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/web_classification/flow.dag.yaml:139]: update here

File: sdk/python/jobs/pipelines/1l_flow_in_pipeline/additional_includes/convert_to_dict.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import json
```
Comment (wangchao1230) [inline sdk/python/jobs/pipelines/1l_flow_in_pipeline/additional_includes/convert_to_dict.py:1]: let's remove this additional includes in sdk example  and make it depends on the cli example files

File: sdk/python/jobs/pipelines/1l_flow_in_pipeline/flow_in_pipeline.ipynb [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    "print(ml_client.compute.get(cluster_name))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2. Load flow as component\n",
+    "\n",
+    "We suppose that there has already been a flow authored with Promptflow SDK/CLI/portal. Then we can load its flow dag yaml as a component like regular component specs."
```
Comment (wangchao1230) [inline sdk/python/jobs/pipelines/1l_flow_in_pipeline/flow_in_pipeline.ipynb:106]: add some link to promptflow repo and doc

File: sdk/python/jobs/pipelines/1l_flow_in_pipeline/web_classification/flow.dag.yaml [hunk 7]
```diff
@@ -0,0 +1,139 @@
+$schema: https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json
```
Comment (wangchao1230) [inline sdk/python/jobs/pipelines/1l_flow_in_pipeline/web_classification/flow.dag.yaml:1]: add a readme saying this example is copied from promptflow repo?

File: sdk/python/jobs/pipelines/1l_flow_in_pipeline/additional_includes/convert_to_dict.py (hunk 5 above)
Comment (elliotzh) [reply sdk/python/jobs/pipelines/1l_flow_in_pipeline/additional_includes/convert_to_dict.py:1]: I haven't done this as the relative path will be extremely long. hmm, will try to optimize this

File: cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/pipeline.yml (hunk 1 above)
Comment (elliotzh) [reply cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/pipeline.yml:4]: it's copied from previous sample in azure-ai-ml, should be removed

File: cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/README.md
Comment (elliotzh) [reply cli/jobs/pipelines-with-components/flow_in_pipeline/1a_flow_in_pipeline/README.md:1]: Noted that `readme.py` will bring huge changes now, updating many files and adding many files. Some people has updated workflow manually, other guys have updated `readme.py` and filter out files they don't want to change.

1. if we want to continuously add examples in this repo and maintain `readme.py`, we will need [a standalone work item](https://msdata.visualstudio.com/Vienna/_workitems/edit/2743872) of around 2 person/day to track this.
2. next time we enable such script, it must be run in ci, blocking PRs if any changes found after rerun the scripts.

File: sdk/python/setup.sh [hunk 8]
```diff
@@ -20,4 +20,8 @@ pip install azure-ai-ml
 # https://docsupport.blob.core.windows.net/ml-sample-submissions/1905732/azure_ai_ml-1.0.0-py3-none-any.whl
 # </az_ml_sdk_test_install>
 
+# <connection_manage_install>
+pip install promptflow
+# </connection_manage_install>
+
```
Comment (kdestin) [inline sdk/python/setup.sh:26]: Any chance this can be moved into the notebook that depends on this?

**Review by kdestin — DISMISSED** [review]

**Review by brynn-code — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**